src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+public class MetricsEmitter {
+    private final ScheduledExecutorService ses;
+    private static final int PERIODIC_PUBLISH_INTERVAL = 86_400;
+    private ClientDeviceAuthMetrics metrics;
+
+    /**
+     * Constructor for the MetricsEmitter.
+     *
+     * @param ses
+     * @param metrics
+     */
+    @Inject
+    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics) {
```

Comment:
Follow the pattern that I linked you in order to read the emission rate from the configuration.

This should emit once an hour by default. It is then aggregated (not by you) once a day and sent to the cloud.

Reply:
@sameerzuberi is the config stuff being addressed in this PR?

Reply:
How could I go about emitting on a timer without the executor service? For context, I was following the format in the TelemetryAgent here: [https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/221c28bde96ee7c9aec853ed81fffeee0376ae43/src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java#L190](url)

Reply:
Ok I see that this is separate from the main metrics class. So that's good. But I actually don't think this is needed at all. It's just another layer that doesn't really provide value.

Right now we're kinda gluing pieces together in the service entry class. It's a bit of a dumping ground, but I'd honestly rather concentrate how everything gets hooked up in that one central place.

E.g. add a `initializeMetrics` startup method or something.

https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/8682bcb81b23022fe43a95f18fe67229d8acbad8/src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java#L117

Also take a look at https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/main/src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

There are some commonalities between these two use cases. Ideally we'd actually just have a scheduler as part of infra, and can just ask it to periodically do things rather than re-creating the wheel

Reply:
I'll take a look into this and adjust, thank you!

Reply:
Chatted offline. There are some complexities with the CertRefresh logic that aren't present here, so we'll keep it simple for now